### PR TITLE
Added new board sunton_esp32_2432S024C

### PR DESCRIPTION
**Pull request by oschettler**
Dear circuitpython team,

I have added a new board. This board is a [Sunton ESP32_2432S024C](https://www.aliexpress.com/item/1005005865107357.html). I have no affiliation with this vendor, but am a user of the board.

The board is one of the Cheap Yellow Display ("CYD") variety. It has a 320x240 pixel screen (ILI9341) with capacitive touch (CST820), an SD card slot, an RGB LED, and a light sensor.

There is some documentation on the board and some working configurations as well:

* Dokumentation (as [ZIP](http://pan.jczn1688.com/directlink/1/ESP32%20module/2.4inch_ESP32-2432S024.zip)), linked from the Aliexpress site above
* a [library for Micropython](https://github.com/jtobinart/Micropython_CYDc_ESP32-2432S024C). The source coude in cydc.py contains a list of all the pins.
* a [board definition](https://github.com/rzeldent/platformio-espressif32-sunton/blob/main/esp32-2432S024C.json) for platformio. This is part of a project to provide board descriptions for all the (yellow) Sunton boards.

### CODE REVIEW

**Comment by dhalbert** [comment]
Hi - there are multiple similar Sunton boards. There is 
- https://circuitpython.org/board/sunton_esp32_2432S032C/ -- 320x240, 3.2", uses an ST7789
- https://circuitpython.org/board/sunton_esp32_2432S028/ - 320x240, 2.8", uses an ILI9341

Is this board pretty much the same as the second one, except for the display screen size?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 repos:
 -   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.0.1
+    rev: v5.0.0
```
Comment (dhalbert) [inline .pre-commit-config.yaml:9]: Could I ask why you changed this? It's probably a good idea, but maybe let's do it in a different PR.

**Review by dhalbert — CHANGES_REQUESTED** [review]
This looks good. I looked at the other boards, as you mentioned. It seems a bit crazy that the boards all use different combinations of display controllers and touch controllers, but they may use the parts that they can get.

One Q re pre-commit version.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (oschettler) [reply .pre-commit-config.yaml]: You are right. There was a warning about a soon-to-be-deprecated naming and I applied it a little too fast. 
I have added a commit to remain at 4.0.1.

**Review by dhalbert — APPROVED** [review]
Thanks! Could you also submit a PR to https://github.com/adafruit/circuitpython-org to describe the board? Guide here: https://learn.adafruit.com/how-to-add-a-new-board-to-the-circuitpython-org-website

**Comment by oschettler** [comment]
The [PR to circuitpython-org](https://github.com/adafruit/circuitpython-org/pull/1555)

**Comment by RetiredWizard** [comment]
The earlier draft PR has a few pin/feature differences that could be differences in unique builds of the devices. The draft also makes more of the touch interface visible through `board` attributes.

**Comment by gitcnd** [comment]
There are dozens of variations of these boards - all the details are nicely spelled out here:
https://github.com/rzeldent/platformio-espressif32-sunton/blob/main/esp32-8048S043R.json

Instructions for getting many of them working with LVGL can be found here too: https://github.com/lvgl-micropython/lvgl_micropython/discussions
